curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md

```diff
@@ -21,435 +21,501 @@ Placeholder question
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Suppose you are designing a database to hold data about books. You need to keep track of title, year published, and whether loaned out or not. What SQL data type would you use for each of these attributes?
 
 #### --distractors--
 
-Placeholder distractor 1
+Title: `INTEGER`, PublicationYear: `DATE`, IsAvailable: `BOOLEAN`
 
 ---
 
-Placeholder distractor 2
+Title: `VARCHAR`, PublicationYear: `INTEGER`, IsAvailable: `TEXT`
 
 ---
 
-Placeholder distractor 3
+Title: `CHAR(100)`, PublicationYear: `INTEGER`, IsAvailable: `VARCHAR(3)`
 
 #### --answer--
 
-Placeholder answer
+Title: `VARCHAR`, PublicationYear: `INTEGER`, IsAvailable: `BOOLEAN`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What SQL clause is used to filter data based on a specified condition within a subset of rows that have been grouped by a specified condition?
 
 #### --distractors--
 
-Placeholder distractor 1
+`WHERE`
 
 ---
 
-Placeholder distractor 2
+`ORDER BY`
 
 ---
 
-Placeholder distractor 3
+`LIMIT`
 
 #### --answer--
 
-Placeholder answer
+`HAVING`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What command would you use to redirect the errors of a command to a file named "errors.log"?
 
 #### --distractors--
 
-Placeholder distractor 1
+`2>&1 > errors.log`
 
 ---
 
-Placeholder distractor 2
+`> errors.log`
 
 ---
 
-Placeholder distractor 3
+`>> errors.log`
 
 #### --answer--
 
-Placeholder answer
+`2> errors.log`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You are working on a Linux system and need to write a bash script that performs these tasks:
+1. List all files and folders, including hidden ones, within the "/home/user" directory.
+2. Copy all ".txt" files from the "/home/user" directory to a folder named "text_backup" within the current directory.
+3. Delete all files with the ".tmp" extension in the "/home/user" directory, asking for confirmation before deleting each file.
 
 #### --distractors--
 
-Placeholder distractor 1
+```
+ls -a /home/user
+cp *.txt text_backup/
+rm -i *.tmp
+```
 
 ---
 
-Placeholder distractor 2
+```
+ls -a /home/user
+cp *.txt /home/user/text_backup/
+rm -rf /home/user/*.tmp
+```
 
 ---
 
-Placeholder distractor 3
+```
+ls -a /home/user
+cp /home/user/*.txt /home/user/text_backup/
+rm -i *.tmp
+```
+
 
 #### --answer--
 
-Placeholder answer
+```
+ls -a /home/user
+cp /home/user/*.txt text_backup/
+rm -i /home/user/*.tmp 
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You are designing a database for a music store. You need to keep track of albums and their artists. You have two tables:
+
+**Table: Albums**
+
+| Column Name | Data Type  | Constraints |
+|-------------|------------|-------------|
+| AlbumID     | INT        | PRIMARY KEY |
+| AlbumTitle   | VARCHAR(255) | NOT NULL   |
+| ReleaseYear  | INT        |             |
```

Comment:
````suggestion
```md
| Column Name | Data Type  | Constraints |
|-------------|------------|-------------|
| AlbumID     | INT        | PRIMARY KEY |
| AlbumTitle   | VARCHAR(255) | NOT NULL   |
| ReleaseYear  | INT        |             |
```
````